optuna/multi_objective/trial.py

```diff
@@ -102,30 +102,24 @@ def suggest_int(
         return self._trial.suggest_int(name, low, high, step=step, log=log)
 
     @overload
-    def suggest_categorical(self, name: str, choices: Sequence[None]) -> None:
-        ...
```

Comment:
What about using `pass` instead of `...`?

Reply:
I have no strong opinion on this topic but it is reasonable to ignore `E704` in this case as mentioned in https://github.com/psf/black/issues/3887 and [`PEP8`](https://peps.python.org/pep-0008/#other-recommendations). I think it is possible to put `pass` since `NotImplementedError` raises whatever we write in the body when function with `@overload` is called.

Reply:
IMO, changing the implementation is out of scope of this PR, since this PR aims just to introduce black 2024.